independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java

```diff
@@ -247,6 +248,10 @@ public void handle(Throwable failure) {
         }, new Consumer<>() {
             @Override
             public void accept(Throwable event) {
+                // set some properties to prevent NPEs down the chain
+                requestContext.setResponseHeaders(new MultivaluedTreeMap<>());
```

Comment:
Is this always the first handler invoked? Curious if the response headers could be overwritten if they were set previously

Reply:
It doesn't really matter if it gets overridden. All that matters in this case is that it's not null